packages/eslint-plugin/src/rules/pipe-prefix.ts

```diff
@@ -0,0 +1,113 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+import { PIPE_CLASS_DECORATOR } from '../utils/selectors';
+import {
+  getClassName,
+  getDecoratorPropertyValue,
+  isLiteral,
+  isTemplateLiteral,
+  SelectorValidator,
+} from '../utils/utils';
+
+type Options = [
+  {
+    prefixes: string[];
+  },
+];
+export type MessageIds = 'pipePrefix';
+export const RULE_NAME = 'pipe-prefix';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'layout',
+    docs: {
+      description: `Enforce consistent prefix for pipes.`,
+      category: 'Stylistic Issues',
+      recommended: false,
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          prefixes: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+            minimum: 1,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      pipePrefix: `The name of the Pipe decorator of class "{{className}}" should start with prefix "{{prefixes}}", however its value is "{{propName}}"`,
```

Comment:
I know this is the message we currently have in `Codelyzer`, but I think that the class name and the name (prefix) of `Pipe` a bit unnecessary because, as I mentioned in https://github.com/mgechev/codelyzer/issues/778#issuecomment-467233675 and in other issues, ESLint already returns the exact line with the selector and class name. I guess, in this case, the only relevant information are the allowed prefixes. I suggest something like this:

```suggestion
      pipePrefix: '@Pipe's name should be prefixed by {{prefixes}}',
```

Btw we have [a function in codelyzer's source](https://github.com/mgechev/codelyzer/blob/bd2a7a8160efd91395ca6230e621ca9d862574a1/src/noInputPrefixRule.ts#L49-L60) to "convert" an `array` to human-readable.

Reply:
Your suggestion make sense to me. 👍  As you mentioned, the "convert" function has added to the utils.ts file to make it could be shared across the rule implementation.

Reply:
Nice!